### PR TITLE
remove lib/chain

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

- Remove lib/chain module, a module used for chaining lodash utilities.
- Remove chain from all modules which depend on it, replacing lodash utilities used with chain with native JS.

## What is the value of this and can you measure success?

- Replaces soon to be replaced lodash utilities with native js, and remove need for chain module.

## Does this affect other platforms - Amp, Apps, etc?

No

## Screenshots

N/A

## Tested in CODE?

No

### CODE REVIEW

File: static/src/javascripts-legacy/projects/common/modules/discussion/comment-count.js [hunk 1]
```diff
@@ -54,7 +46,7 @@ define([
     }
 
     function getContentIds(indexedElements) {
-        return chain(indexedElements).and(keys).and(uniq).and(sortBy).join(',').value();
+        return Object.keys(indexedElements).sort().join(',');
```
Comment (SiAdcock) [inline static/src/javascripts-legacy/projects/common/modules/discussion/comment-count.js]: We no longer care if unique?

File: static/src/javascripts-legacy/projects/common/modules/onward/history.js [hunk 2]
```diff
@@ -328,37 +326,40 @@ define([
             page = collapsePath(pageConfig.pageId),
             isFront = false;
 
-        chain(pageMeta).and(reduceRight, function (tagMeta, tag) {
-                var tid = collapsePath(firstCsv(pageConfig[tag.tid])),
-                    tname = tid && firstCsv(pageConfig[tag.tname]);
+        var meta = pageMeta.reduceRight(function (tagMeta, tag) {
```
Comment (SiAdcock) [inline static/src/javascripts-legacy/projects/common/modules/onward/history.js]: Had no idea there was a `reduceRight` in JS 🎉

**Comment by SiAdcock** [comment]
OMG this is amazing 😲

File: static/src/javascripts-legacy/projects/common/modules/discussion/comment-count.js (hunk 1 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/discussion/comment-count.js]: lodash keys would've returned an array composed of the key names from the indexedElements object, keys on an object should be unique anyway so I thought this was unnecessary?

Comment (SiAdcock) [inline static/src/javascripts-legacy/projects/common/modules/discussion/comment-count.js]: Ha! Good catch! 👍

Comment (sndrs) [inline static/src/javascripts-legacy/projects/common/modules/discussion/comment-count.js]: joy of tests 🏝

**Comment by sndrs** [comment]
seems like a good test in code would be useful here?

**Review by SiAdcock — APPROVED** [review]
Assuming all looks good on CODE 👍

**Comment by GHaberis** [comment]
@sndrs @SiAdcock yep I'm going to give this a whirl on code first to check on everything!

**Comment by GHaberis** [comment]
I tested on CODE and found 1 issue with the search box on the weather widget, I've fixed this and everything else seems fine so will merge pending Team City checks.

**Pull request merged**